subprojects/base-services/src/main/java/org/gradle/internal/operations/notify/BuildOperationStartedNotification.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.operations.notify;
+
+public interface BuildOperationStartedNotification {
```

Comment:
I think we should leave it weak and add back the types on the receiving side. This would prevent that interface from being part of the cross version contract. Since we are decoupling, we should go all out.